src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp

```diff
@@ -1194,7 +1199,48 @@ WaveformAppearance::WaveformAppearance(WaveTrack &track)
 
 WaveformAppearance::~WaveformAppearance() = default;
 
+void WaveformAppearance::OnNewClip(WaveClip &clip, bool)
+{
+   WaveClipWaveformCache::Get(clip).SetColorIndex(mColorIndex);
```

Comment:
My guess is that there will be an extra call to `SetColorIndex` during deserialization: one from `OnNewClip`, and second in `HandleXMLAttribute`, right?

Reply:
Yes, does it matter much?

It would matter in case a file serialized wave color without clip color, but then the clip colors would still be initialized.

I don't think there are any such legacy files, though.